handlers/mesh.go

```diff
@@ -63,3 +69,14 @@ func MeshGraph(
 		respond(w, code, payload)
 	}
 }
+
+func filterAccessibleControlPlanes(ctx context.Context, namespaceService business.NamespaceService, mesh *models.Mesh) {
+
+	for i, cp := range mesh.ControlPlanes {
+		// Check if the user is able to access to the control plane
+		_, err := namespaceService.GetClusterNamespace(ctx, cp.IstiodNamespace, cp.Cluster.Name)
+		if err != nil && errors.IsForbidden(err) {
+			mesh.ControlPlanes = slices.Delete(mesh.ControlPlanes, i, i+1)
+		}
+	}
```

Comment:
The `mesh.ControlPlanes` slice is being ranged over here and then inside the loop it is also being modified. I'm not sure but this might cause weird side effects. To be safe, you could either build a new slice with just the filtered elements or else keep track of all the indexes to be deleted inside the `range mesh.ControlPlanes` loop and then call `slices.Delete` on them afterward outside the `range mesh.ControlPlanes` loop.

Reply:
Right, maybe that can cause issues